### ISSUE TITLE
HTML Import causes collapsed output cells on Chrome.

### ISSUE THREAD

**Issue by deedubbu** [jupyter/notebook#429]
Using the html magic to perform an HTML import causes output cells to collapse on reload of the notebook in Chrome (Firefox and Safari are ok). In my notebook, I've created a simple HTML file that I am then importing into a notebook, using the native HTML import functionality available in Chrome. Running the cell and reloading the page causes any cells that had previously been run to be displayed collapsed.
1. Create a simple html file in the notebook
   New -> Text File (Set name as filetoimport.html)
   Contents:
   `
   <script>
       console.log('HTML Import succeeded.');
   </script>
   `
2. Create a new IPython notebook and add following cells:
   <img width="793" alt="notebook_cells" src="https://cloud.githubusercontent.com/assets/782384/9862933/cb3c6e0a-5b01-11e5-858d-b487f5125bcf.png">
3. Run all the cells
4. Save the notebook.
5. Reload the notebook.
6. Notice that cell 1 displays correctly but cells 2 and 3 are collapsed and do not show the contents.
   <img width="956" alt="notebook_cells_collapsed" src="https://cloud.githubusercontent.com/assets/782384/9860759/c9f61f54-5af3-11e5-8eb0-8731b117d568.png">

I've tried the above scenario on Safari and Firefox with the additional step of including the web-components polyfill and those browsers do not exhibit the same issue.

I did a brief comparison in the styling of the DOM between when the cells display correctly and when they don't and found that some of the inline element styling in the code-mirror DOM elements is different.

Here is the inline styling when cells are displayed correctly:
<img width="706" alt="style_cells_displayed" src="https://cloud.githubusercontent.com/assets/782384/9860777/e26d866c-5af3-11e5-838a-9e2f878e18ca.png">

And here is what it looks like when cells get collapsed on reload:
<img width="716" alt="style_cells_collapsed" src="https://cloud.githubusercontent.com/assets/782384/9860791/f0747d60-5af3-11e5-8fb7-737f87481835.png">

**Comment by Carreau** [comment]
unfortunately this looks like a chrome rendering bug. As soon as you trigger a redraw (open dev tools, or resize the window most of the time) the css computation is done correctly. So I doubt there is anything that we can do. 


@rgbkrk and @jdfreder you have played with webComponents. Ideas ?

**Comment by jdfreder** [comment]
I haven't played with webComponents in the notebook specifically, and @rgbkrk has much more experience with webComponents, but my gut tells me your initial diagnosis is correct.  This seems like a chrome bug.

**Comment by Carreau** [comment]
I'm not sure there is anything we can do, bump as 4.x

**Comment by rgbkrk** [comment]
What do you think about this one  @lbustelo?

**Comment by lbustelo** [comment]
I can confirm the behavior that @deedubbu describes above. Here is a bit more information of what I noticed.

I'm going to refer as the `link cell` the cell that contains the  `<link rel='import'>`

1. Cells below the `the link cell` are affected. Cells above the `link cell` seem to render fine.
2. It seems that chrome holds of rendering any cells after the `link cell`. I can see this by placing a breakpoint at `CodeCell.prototype.create_element`. When there is no `link cell`, you can see the browser rendering the cell after each iteration call to `create_element`. When there is a `link cell', you can see the 'link cell' and any cell above render after completing the call to `create_element`, but the cells afters don't show up until later and without `min-height` set to the `CodeMirror-scroll` element.

It does seem like Chrome is making changes to its rendering when a `<link rel='import'>` is added. Any attempt to fix this would probably include experimenting with rendering each cell on separate event loop.

**Comment by minrk** [comment]
Possibly related to this is [this notebook](http://nbviewer.jupyter.org/github/garth-wells/fenics-x3dom/blob/master/DOLFIN_X3DOM.ipynb) not properly displaying its output (the last should have color) in nbconvert output, but working in the live notebook.

**Comment by martinal** [comment]
Reported here now:
https://bugs.chromium.org/p/chromium/issues/detail?id=663288

**Comment by martinal** [comment]
Chromium devs have looked into the dolfin x3dom notebook example (see post from @minrk above) which is a bug in the javascript. They state this issue is separate and request a new issue opened with an updated reproducible case if they are to look into it.

**Comment by takluyver** [comment]
Is there anything for us to fix on this?

**Comment by martinal** [comment]
The notebook min posted above Apr 13 is not relevant here, that was a bug in x3dom which is now fixed.

**Comment by takluyver** [comment]
OK, then it sounds like there isn't an issue in notebook to be fixed here. If I'm wrong about this, we can reopen it.

**Comment by lbustelo** [comment]
@takluyver I can still recreate on try.jupyter.org using Chrome 55. It may be that there is no way of fixing this in the Notebook.

<img width="1235" alt="screen shot 2017-02-03 at 3 58 20 pm" src="https://cloud.githubusercontent.com/assets/2474841/22610161/bffebc6a-ea29-11e6-8b94-c8887670cb8b.png">

**Comment by takluyver** [comment]
Right, my understanding of the consensus was that it's an issue in Chrome.

**Comment by lbustelo** [comment]
Browsers are full of bugs! Not sure it is ok to close this even if the fix will eventually land on chrome. This is very prevalent when using declarative widgets and is related to how the notebook loads cells on refresh. 

We should either keep open while waiting for chrome to fix or investigate how to remediate in notebook cell loading code.

**Comment by takluyver** [comment]
Ok
